File: 01_ollama_client.py
import time
import ollama

client = ollama.Client(host="http://localhost:11434")
model = "llama2"
prompt = "What is Python?"

start_time = time.time()
try:
    response = client.generate(model=model, prompt=prompt)
    end_time = time.time()
    elapsed_time = end_time - start_time

    print("\nResponse from Ollama:")
    print(response.response)
    print(f"\nResponse time: {elapsed_time:.2f} seconds")

except Exception as e:
    print("Error communicating with Ollama:", str(e))

File: nodetest/01_ollama_client.js
import ollama from 'ollama'

const model = "llama2"
let prompt = "Hello"

let start_time = Date.now();

try {
    const response = await ollama.chat({
        model: model,
        messages: [{ role: 'user', content: prompt }]
    })
    let end_time = Date.now();
    let elapsed_time = end_time - start_time;

    console.log("Response from "+ model + ":")
    console.log(response.message.content)
    console.log(`Response time: ${elapsed_time / 1000} seconds`)
} catch (e) {
    console.log(e);
}